### PR TITLE
Refactor ProcessManagerImpl

### PR DESCRIPTION
**Pull request by antonrohr**
This adds a dtor to ProcessManagerImpl that calls Shutdown.

### CODE REVIEW

File: OrbitClientServices/ProcessManager.cpp [hunk 1]
```diff
@@ -23,6 +24,9 @@ class ProcessManagerImpl final : public ProcessManager {
  public:
   explicit ProcessManagerImpl(const std::shared_ptr<grpc::Channel>& channel,
                               absl::Duration refresh_timeout);
+  ProcessManagerImpl(ProcessManagerImpl&&) = default;
+  ProcessManagerImpl& operator=(ProcessManagerImpl&&) = default;
+  ~ProcessManagerImpl() override { Shutdown(); }
```
Comment (dimitry-) [inline OrbitClientServices/ProcessManager.cpp:29]: nit: wanted to do that but while you are here maybe rename Shutdown() to ShutdownAndWait()?

Comment (dimitry-) [reply OrbitClientServices/ProcessManager.cpp:29]: Also are we sure it does not throw bad_alloc? Maybe make it nothrow as well.

Comment (antonrohr) [reply OrbitClientServices/ProcessManager.cpp]: I am not completely sure if I understood the nothrow part. I made the function noexcept now. Please take a look

File: OrbitClientServices/ProcessManager.cpp [hunk 2]
```diff
@@ -85,7 +89,7 @@ void ProcessManagerImpl::Start() {
   worker_thread_ = std::thread([this] { WorkerFunction(); });
 }
 
-void ProcessManagerImpl::Shutdown() {
+void ProcessManagerImpl::ShutdownAndWait() noexcept {
   shutdown_mutex_.Lock();
   shutdown_initiated_ = true;
   shutdown_mutex_.Unlock();
```
Comment (dimitry-) [inline OrbitClientServices/ProcessManager.cpp:95]: ok, this is not safe - worker_thread_.join() below can throw an exception. Maybe we need JoinThreadOrDie function here which does not throw.

Comment (beckerhe) [reply OrbitClientServices/ProcessManager.cpp:95]: I think it's fine to just swallow the exception in that case. Something like:
```
try {
worker_thread_.join();
catch (const std::exception&) {}
```

@dimitry- WDYT?

Comment (dimitry-) [reply OrbitClientServices/ProcessManager.cpp:95]: well, I think we kind of want to add FATAL in exception handling part at least, preferably checking for conditions before too.

Comment (antonrohr) [reply OrbitClientServices/ProcessManager.cpp:95]: I did not add another JoinThreadOrDie function, but catch the potential exception there. ptal

**Review by dimitry- — APPROVED** [review]

**Pull request merged**